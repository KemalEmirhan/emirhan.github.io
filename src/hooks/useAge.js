const BIRTH_YEAR = 1994;

export const useAge = () => {
    return ((new Date()).getFullYear() - BIRTH_YEAR) - 1;
}