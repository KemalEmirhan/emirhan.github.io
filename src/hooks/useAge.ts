const BIRTH_YEAR = 1994;

const useAge = (): number => {
    return ((new Date()).getFullYear() - BIRTH_YEAR) - 1
}

export default useAge;
