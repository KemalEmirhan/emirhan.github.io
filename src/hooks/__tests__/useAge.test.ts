import { describe, it, expect, vi, beforeAll, afterAll } from 'vitest';
import useAge from '../useAge';
import { renderHook } from '@testing-library/react';

// Mock the BirthDateSchema to control the birth date for testing
vi.mock('@/schemas', () => ({
  BirthDateSchema: {
    parse: vi.fn(() => new Date(1994, 7, 13)), // August 13, 1994
  },
}));

describe('useAge', () => {
  beforeAll(() => {
    vi.useFakeTimers();
  });

  afterAll(() => {
    vi.useRealTimers();
  });

  it('should calculate age correctly before birthday', () => {
    // Mock current date to be July 15, 2024 (before birthday)
    vi.setSystemTime(new Date(2024, 6, 15)); // July 15, 2024

    const result = renderHook(() => useAge()).result.current;
    expect(result).toBe(29);
  });

  it('should calculate age correctly on birthday', () => {
    // Mock current date to be August 13, 2024 (on birthday)

    vi.setSystemTime(new Date(2024, 7, 13)); // August 13, 2024

    const result = renderHook(() => useAge()).result.current;
    expect(result).toBe(30);
  });

  it('should calculate age correctly after birthday', () => {
    // Mock current date to be September 15, 2024 (after birthday)
    vi.setSystemTime(new Date(2024, 8, 15)); // September 15, 2024

    const result = renderHook(() => useAge()).result.current;
    expect(result).toBe(30);
  });
});
