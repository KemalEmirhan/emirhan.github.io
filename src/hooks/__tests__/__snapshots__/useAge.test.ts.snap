// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useAge > should calculate age correctly for current year 1`] = `30`;

exports[`useAge > should calculate age correctly for next year 1`] = `31`;

exports[`useAge > should calculate age correctly for previous year 1`] = `29`;
